ravedude/src/board.rs

```diff
@@ -1,439 +1,89 @@
-        None // The TrinketPro does not have USB-to-Serial.
-    }
-}
-
-struct Trinket;
-
-impl Board for Trinket {
-    fn display_name(&self) -> &str {
-        "Trinket"
-    }
-
-    fn needs_reset(&self) -> Option<&str> {
-        Some("Reset the board by pressing the reset button once.")
-    }
-
-    fn avrdude_options(&self) -> avrdude::AvrdudeOptions {
-        avrdude::AvrdudeOptions {
-            programmer: "usbtiny",
-            partno: "attiny85",
-            baudrate: None,
-            do_chip_erase: true,
-        }
-    }
-
-    fn guess_port(&self) -> Option<anyhow::Result<std::path::PathBuf>> {
-        None // The Trinket does not have USB-to-Serial.
-    }
-}
-
-struct Nano168;
-
-impl Board for Nano168 {
-    fn display_name(&self) -> &str {
-        "Nano Clone (ATmega168)"
-    }
-
-    fn needs_reset(&self) -> Option<&str> {
-        None
-    }
-
-    fn avrdude_options(&self) -> avrdude::AvrdudeOptions {
-        avrdude::AvrdudeOptions {
-            programmer: "arduino",
-            partno: "atmega168",
-            baudrate: Some(19200),
-            do_chip_erase: false,
+                if let Some(inherit) = board_config.inherit.as_deref() {
+                    let base_board = get_board(Some(inherit))?.board_config.unwrap();
+                    board.board_config = Some(board.board_config.take().unwrap().merge(base_board));
+                }
+            } else if let Some(board_name) = board.general_options.board.as_deref() {
+                let base_board = get_board(Some(board_name))?.board_config.unwrap();
+                board.board_config = Some(base_board);
+            }
+            board
         }
-    }
-
-    fn guess_port(&self) -> Option<anyhow::Result<std::path::PathBuf>> {
-        Some(Err(anyhow::anyhow!("Not able to guess port")))
-    }
+    })
 }
 
-struct ArduinoDuemilanove;
+#[cfg(test)]
+mod tests {
+    use super::get_all_boards;
 
-impl Board for ArduinoDuemilanove {
-    fn display_name(&self) -> &str {
-        "Arduino Duemilanove"
-    }
-
-    fn needs_reset(&self) -> Option<&str> {
-        None
-    }
+    #[test]
+    fn validate_board_list() -> anyhow::Result<()> {
+        let all_boards = get_all_boards()?;
 
-    fn avrdude_options(&self) -> avrdude::AvrdudeOptions {
-        avrdude::AvrdudeOptions {
-            programmer: "arduino",
-            partno: "atmega328p",
-            baudrate: Some(57600),
-            do_chip_erase: true,
+        for board in all_boards.values() {
+            assert!(board.name.is_some());
+            assert!(board.inherit.is_none());
+            assert!(board.reset.is_some());
+            assert!(board.avrdude.is_some());
+            let avrdude = board.avrdude.as_ref().unwrap();
+            assert!(avrdude.programmer.is_some());
+            assert!(avrdude.partno.is_some());
+            assert!(avrdude.baudrate.is_some());
+            assert!(avrdude.do_chip_erase.is_some());
```

Comment:
Bonus points for making these asserts show what board failed the test :)  You can do it like this:

```rust
for (name, board) in all_boards.items() {
    assert!(board.name.is_some(), "Board {name:?} is missing a `name`");
    assert!(board.inherit.is_none(), "Board {name:?} has illegal `inherit` key");
```

Reply:
Done :+1: